pints/_mcmc/__init__.py

```diff
@@ -774,14 +816,25 @@ def run(self):
             if self._log_to_screen:
                 print(halt_message)
 
-        # Store generated chains in memory
-        if self._chains_in_memory:
-            self._samples = samples
-
         # Store evaluations in memory
         if self._evaluations_in_memory:
             self._evaluations = evaluations
 
+        if self._chains_in_memory:
+
+            # Inverse transform to model space if transform is provided
+            if self._transform:
+                n_c, n_s, n_p = samples.shape
+                samples_user = np.zeros((n_c, n_s, n_p))
```

Comment:
I _think_ that the code in the constructor could actually cope with the situation where n_p isn't the same in the world and search space. At least, I was about to re-order some checks until I figured this was the case, and that you might have done it on purpose.

If so... or just if we want to have the option of having an n_p change, then this code should use `np.zeros((n_c, n_s, original-log-pdf.n_parameters()))` or something similar.

If we think changing `n_p` is a terrible idea, then maybe this code can be optimised by modifying the samples array during the loop, instead of creating a full copy? (so e.g. `samples[c, :, :] = self._transform...`)

Reply:
Most memory-efficient solution would still be to just transform before sticking them in `samples` in the first place!
Then the transformation objects become simpler too? (No need for `multiple_to_model`)

Reply:
> What was the reasoning for transforming the whole chain at the end of the sampling routine, rather than when each sample comes in?
> I know we discussed this before at some point...

> chon  2:33 PM
> Yes haha! Because I want “whatever happens within the sampling route stay in the transformed/search space”. More for in case e.g. later if someone needs a method that takes the previous two steps within the sampled chain, and wonder if it’s transformed or detransformed — suspect quite error prone.  (edited) 

> michael  2:35 PM
> Hmmmm, you can't take 2 samples without losing the markov property
> But also we'll never have two-way communication between sampler and controller! sampler doesn't know about controller
> So if you did need 2 previous samples in a sampler, you'd have to store them yourself
> Same way that samplers already maintain a current state and logpdf, instead of requesting that from the controller

Reply:
(with respect to the possibility of changing n_p, that would mean the preallocation would have to use n_p of "world" samples, I guess)

Reply:
Moved transformation to before assigning values to `samples` now. And added a note to samples saying something like "we store only the detransformed parameters if transform is provided", so if anyone later have a look at `samples`, hopefully they will see that message and know that they are dealing with detransformed parameters.

I will also remove `multiple_to_model` soon.

Reply:
Removed `multiple_to_model` methods.